src/lib/components/toggle/QSwitch.svelte

```diff
@@ -0,0 +1,110 @@
+<script lang="ts">
+  import QIconSnippet from "$components/private/QIconSnippet.svelte";
+  import { ripple } from "$helpers";
+  import { isActivationKey } from "$utils";
+  import type { QSwitchProps } from "./props";
+
+  let {
+    value = $bindable(),
+    label = undefined,
+    labelPosition = "right",
+    disabled = false,
+    icons = false,
+    showOnlyCheckedIcon = false,
+    checkedIcon,
+    uncheckedIcon,
+    ...props
+  }: QSwitchProps = $props();
+
+  let qSwitch: HTMLDivElement;
+  let qSwitchInput: HTMLInputElement;
+
+  function toggle() {
+    value = !value;
+  }
+
+  function onclick() {
+    if (!qSwitchInput || disabled) {
+      return;
+    }
+
+    qSwitchInput.focus();
+    toggle();
+  }
+
+  function onkeydown(event: KeyboardEvent) {
+    if (!qSwitch || disabled || !isActivationKey(event) || event.defaultPrevented) {
+      return;
+    }
+
+    event.preventDefault();
+    qSwitchInput.click();
+  }
+
+  Q.classes("q-switch", {
+    bemClasses: {
+      disabled,
+      reversed: labelPosition === "left",
+    },
+    classes: [props.class],
+  });
+
+  Q.classes("q-switch__handle", {
+    bemClasses: {
+      "with-icon": showOnlyCheckedIcon ? !!value : icons || uncheckedIcon,
+    },
+  });
+</script>
+
+<div
+  bind:this={qSwitch}
+  {...props}
+  class="q-switch"
+  {...Q.classes}
+  {onclick}
+  {onkeydown}
+  aria-disabled={disabled || undefined}
+  role="switch"
+  aria-checked={value || undefined}
+  tabindex={disabled ? -1 : 0}
+>
+  <label class="q-switch__inner">
+    <input
+      bind:checked={value}
+      bind:this={qSwitchInput}
+      type="checkbox"
+      disabled={disabled || undefined}
+    />
+    <span class="q-switch__track">
+      <span class="q-switch__touch"></span>
+      <span class="q-switch__handle-container" use:ripple={{ disabled }}>
+        <span class="q-switch__handle" {...Q.classes}>
+          {#if (uncheckedIcon || icons) && !showOnlyCheckedIcon}
```

Comment:
```svelte
{#if (uncheckedIcon || icons)}
  {@render icon(!showOnlyCheckedIcon ? "unchecked" : "checked")}
{/if}
```
Instead of two if blocks?

Reply:
No, I can't do that as the first block checks for `unchecked || icons` and the second checks for `checked || icons`